simpleflow/process/supervisor.py

```diff
@@ -106,10 +126,38 @@ def _handle_graceful_shutdown(signum, frame):
                 signal_name, os.getpid()))
             self.terminate()
 
+        # NB: Function is nested to have a reference to *self*.
+        def _handle_sigchld(signum, frame):
+            """
+            Handles SIGCHLD signal at supervisor process level. By construction this
+            handler is only attached to this process, not worker processes nor deciders.
+            """
+            logger.debug("process: caught signal signal=SIGCHLD, will cleanup zombies "
+                         "from pid={}".format(os.getpid()))
+
+            # cleanup children
+            # NB: here we use psutil's Process() API because it helps us identifying children
```

Comment:
Does this work reliably?

`psutil.Process().children()` is:
```python
                for p in process_iter():
                    try:
                        if p.ppid() == self.pid:
                            # if child happens to be older than its parent
                            # (self) it means child's PID has been reused
                            if self.create_time() <= p.create_time():
                                ret.append(p)
                    except (NoSuchProcess, ZombieProcess):
                        pass
```
Thus it skips zombies (p.ppid() == 1)